### PR TITLE
Banner typescript

### LINKED ISSUE #554: Refactor Notification Banner to Typescript
Create typings for [Notification Banner](https://eds.equinor.com/components/notification-banner/)

Make sure you fork the repository and branch from `typescript` before starting on this issue

### PR DESCRIPTION
**Pull request by nicholasdalhaug**
Resolves #554 

Note that the first commit with renaming enables easy view of what has changed, but it gives failing tests and code between the commits. This pr should be squashed before merge.

### CODE REVIEW

File: libraries/core-react/src/Banner/BannerActions.tsx [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
-import React from 'react'
-import PropTypes from 'prop-types'
+import React, { HTMLAttributes, ReactNode } from 'react'
 import styled from 'styled-components'
 import { banner as tokens } from './Banner.tokens'
 
 const { enabled } = tokens
 
-const StyledBannerActions = styled.div`
+type BannerActionsPlacement = 'bottom' | 'left'
```
Comment (nicholasdalhaug) [inline libraries/core-react/src/Banner/BannerActions.tsx:7]: Not sure if this is a good way to reduce code duplication? Was considering making a types file, but I am not sure what it should look like.

File: libraries/core-react/src/Banner/BannerMessage.tsx [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
+import { Typography  } from '../Typography'
+import { Props as TypographyProps } from '../Typography/Typography'
 
 const StyledBannerMessage = styled(Typography)``
 
-export const BannerMessage = ({ children, ...props }) => {
+type Props = {
+  /** Text content */
+  children: string
+} & Omit<TypographyProps,'children'>
```
Comment (nicholasdalhaug) [inline libraries/core-react/src/Banner/BannerMessage.tsx:11]: Note here the forcing of children to be a string, but also enabling easy props view for the props that will be sent further to typography.

File: libraries/core-react/src/Typography/Typography.tsx [hunk 3]
```diff
@@ -81,7 +81,8 @@ const StyledTypography = styled.p<StyledProps>`
       }
     `}
 `
-type Props = {
+
+export type Props = {
```
Comment (nicholasdalhaug) [inline libraries/core-react/src/Typography/Typography.tsx:85]: Exported to be able to read in `BannerMessage.tsx`, but not exported via `Typography/index.ts`.

**Comment by nicholasdalhaug** [comment]
Tests pass locally. Can you see what is wrong in Azure Pipelines?

File: libraries/core-react/src/Banner/BannerActions.tsx (hunk 1 above)
Comment (mimarz) [reply libraries/core-react/src/Banner/BannerActions.tsx:7]: For now we have defined them in the component they are used. If there are any that are shared across sub-components. We have normally just created a `types.ts`. We have a rollup plugin that extract types for dist, but are anyways planning to have revision on types when we finish rewriting as to better see what can be shared and not.

**Comment by mimarz** [comment]
> Tests pass locally. Can you see what is wrong in Azure Pipelines?

@nicholasdalhaug Seems to be some linting and formatting errors... Do you have eslint or prettier setup in your IDE that are not reading our rules? The prettier errors are not there on `typescript` 🤔

image

**Comment by nicholasdalhaug** [comment]
> @nicholasdalhaug Seems to be some linting and formatting errors... Do you have eslint or prettier setup in your IDE that are not reading our rules? The prettier errors are not there on `typescript` 🤔

You are totally right, my eslint was disabled in vscode. Seems like the tests passed now 😊

**Review by mimarz — APPROVED** [review]
LGTM 👍 

Thanks for your contribution!

**Comment by nicholasdalhaug** [comment]
Could you label the pr 'hacktoberfest-accepted' or use the 'hacktoberfest' topic on the repo? 😊

**Pull request merged**